hack/provision.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+set -eu
+
+if [ "$EUID" -ne 0 ]; then
+  echo "Please run as root"
+  exit
+fi
+
+echo ""
+echo "Provisioning..."
+
+if [ $# -lt 7 ]; then
+  echo "Usage: ./setup.sh <github-user> <github-repo> <actions-runner-token> <actions-runner-name> <consul-bind-ip> <consul-server-ip> [agent|web]"
+  exit 1
+fi
+
+ACTIONS_RUNNER_VERSION=2.278.0
+ACTIONS_RUNNER_USER=github-runner
+ACTIONS_RUNNER_HOME=/srv/github-runner
+ACTIONS_RUNNER_REPO_OWNER=$1
+ACTIONS_RUNNER_REPO_NAME=$2
+ACTIONS_RUNNER_TOKEN=$3
+ACTIONS_RUNNER_NAME=$4
+ACTIONS_RUNNER_PATH="$ACTIONS_RUNNER_HOME/actions-runner"
+ACTIONS_RUNNER_REPO_URL="https://github.com/$ACTIONS_RUNNER_REPO_OWNER/$ACTIONS_RUNNER_REPO_NAME"
+ACTIONS_RUNNER_SYSTEMD_UNIT="actions.runner.$ACTIONS_RUNNER_REPO_OWNER-$ACTIONS_RUNNER_REPO_NAME.$ACTIONS_RUNNER_NAME.service"
+
+CONSUL_VERSION=1.9.6
+CONSUL_BIND_IP=$5
+CONSUL_SERVER_IP=$6
+CONSUL_USER=consul
+CONSUL_HOME=/srv/consul
+CONSUL_CONFIG_PATH="$CONSUL_HOME/consul.d"
+
+ROLE=$7
+if [ "$ROLE" = "agent" ]; then
+  CONSUL_HCL="consul-client.hcl"
+elif [ "$ROLE" = "web" ]; then
+  CONSUL_HCL="consul-server.hcl"
+else
+  echo "Please specify a valid role"
+  exit
+fi
+
+CONSUL_HCL_TEMPLATE_URL="https://raw.githubusercontent.com/trento_project/trento/main/hack/$CONSUL_HCL.template"
```

Comment:
Sorry that I didn't ask this before, but wouldn't it be possible to also use here `$ACTIONS_RUNNER_REPO_OWNER` and `$ACTIONS_RUNNER_REPO_NAME`? I was just testing the script on my fork and had to manually edit this.

Reply:
Not sure about this since it's not gonna work till it's merged because of the hardcoded branch name. Adding an optional branch parameter just for this seems a bit overkill.